make sure to export audio apis in exported applab apps

pcardune opened the pull request.

pcardune commented:
@islemaster looks like this was caused by the changes in #12543. I'm thinking there must be a better way to get the complete list of all commands exposed as global functions, but I don't understand how all the different command configs get hooked up. Do you have any suggestions on that front? While the bug is fixed in this PR, I think the fix is rather brittle.

islemaster commented:
I think there are already a couple of other things that might be missing here - the [dontMarshalApi](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/applab/dontMarshalApi.js) which also gets referenced in applab/dropletConfig.js, and the [makerlab api.js](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/makerlab/api.js) which gets referenced from [makerlab's dropletconfig](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/makerlab/dropletConfig.js) which gets [mixed in to applab's dropletconfig in some cases](https://github.com/code-dot-org/code-dot-org/blob/4a05a9526fcd5b8749691cb90ed2d3fd48e168aa/apps/src/applab/applab.js#L683-L687).

Maybe it would be more reliable to run through the final dropletConfig and look at the unique `parent`s of the various passed commands?  I also think this whole droplet setup is in need of some refactoring anyway - I'd like to see more subsets of commands broken out into small modules like makerlab and the audio commands, but we don't really have a standard approach to that yet.

pcardune commented:
Ok, I refactored things a bit and added tests for all the functions exposed by dropletConfig. The makerlab apis are still unsupported at this point. I haven't played with that stuff at all yet.

islemaster commented:
What's with all the updated localizations in the diffs?

pcardune commented:
accidentally pulled from staging even though this PR was not against staging. Switched the base of this PR to staging and now it looks right.

islemaster commented on apps/test/unit/applab/exportTest.js:
```diff
@@ -0,0 +1,29 @@
+            assert(false, `expected exported globals to include ${block.func}`);
+          }
+        }
+      });
```
This is pretty sweet - requiring internal consistency between the `dropletConfig` and the exported globals.  But there's something I don't understand about the export process here - do these necessarily need to be global?  In `dropletConfig` most entries have a `parent` object where they'll supposedly find the method they're looking for, without it being on `window` AFAIK.

pcardune replied:
exported apps do not run through our interpreter. Instead, you just end up with a script tag pointing to a file with the student's js code. As a result, and functions that the student's code might call to need to be on `window`.

islemaster replied:
Got it.  _Great_ test then!

islemaster commented on apps/src/applab/export.js:
```diff
@@ -0,0 +1,7 @@
+import * as dontMarshalApi from './dontMarshalApi';
+
+export function getExportedGlobals() {
+  return Object.assign({}, executors, dontMarshalApi, api);
```
:+1: To isolating this weirdness into its own file.  It'd be nice to have a comment explaining what this is for though - it's not obvious to me in isolation.

islemaster commented:
LGTM!
